src/components/sections/judge/JudgeContests.tsx

```diff
@@ -0,0 +1,207 @@
+    useSearchParamsNumericState('page', 1)
+  const [sortParams, {getSortParamsUpdaters: updateSortSearchParams}] =
+    useSortingSearchParams(JudgeContestSorting)
+  const [contestStatus, {getSearchParamsUpdater: updateStatusSearchParams}] =
+    useSearchParamsEnumArrayState('status', ContestStatus)
+
+  const {data: contests, isLoading} = useGetJudgeContests({
+    type: contestType,
+    pageParams: {
+      limit: JUDGE_CONTESTS_PAGE_SIZE,
+      offset: (page - 1) * JUDGE_CONTESTS_PAGE_SIZE,
+    },
+    sort: sortParams,
+    status: contestStatus,
+  })
+
+  const liveContests = useMemo(
+    () =>
+      contests?.data.filter(
+        (contest) =>
+          DateTime.fromJSDate(contest.startDate) < DateTime.now() &&
+          DateTime.fromJSDate(contest.endDate) > DateTime.now(),
+      ),
+    [contests],
+  )
+
+  const futureContests = useMemo(
+    () =>
+      contests?.data.filter(
+        (contest) => DateTime.fromJSDate(contest.startDate) > DateTime.now(),
+      ),
+    [contests],
+  )
+
+  const pastContests = useMemo(
+    () =>
+      contests?.data.filter(
+        (contest) => DateTime.fromJSDate(contest.endDate) < DateTime.now(),
+      ),
+    [contests],
+  )
+
+  const currentContests = useMemo(() => {
+    switch (contestType) {
+      case ContestOccurence.PRESENT:
+        return liveContests
+      case ContestOccurence.FUTURE:
+        return futureContests
+      case ContestOccurence.PAST:
+        return pastContests
+    }
+  }, [contestType, liveContests, futureContests, pastContests])
+
+  const pastCount = contests?.pageParams.pastCount
+  const liveCount = contests?.pageParams.liveCount
+  const futureCount = contests?.pageParams.futureCount
+
+  const currentCount = useMemo(() => {
+    switch (contestType) {
+      case ContestOccurence.PRESENT:
+        return liveCount
+      case ContestOccurence.PAST:
+        return pastCount
+      case ContestOccurence.FUTURE:
+        return futureCount
+    }
+  }, [contestType, liveCount, pastCount, futureCount])
+
+  const getStatusOptions = useMemo(() => {
+    switch (contestType) {
+      case ContestOccurence.PAST:
+        return selectOptions.contestStatus.filter(
+          (status) =>
+            status.value === ContestStatus.PENDING ||
+            status.value === ContestStatus.FINISHED ||
+            status.value === ContestStatus.APPROVED,
+        )
+      case ContestOccurence.PRESENT:
+        return selectOptions.contestStatus.filter(
+          (status) =>
+            status.value === ContestStatus.APPROVED ||
+            status.value === ContestStatus.IN_REVIEW,
+        )
+      case ContestOccurence.FUTURE:
+        return selectOptions.contestStatus.filter(
+          (status) =>
+            status.value === ContestStatus.IN_REVIEW ||
+            status.value === ContestStatus.REJECTED ||
+            status.value === ContestStatus.APPROVED,
+        )
+    }
+  }, [contestType])
```

Comment:
I like it better when the definitions are separate from logic, e.g. to move them outside of the component into a map:

```
{
  [ContestOccurence.PAST]: [ContestStatus.PENDING, ContestStatus.FINISHED, ContestStatus.APPROVED],
  [ContestOccurence.PRESENT]:
  ...
}
```
and then just use it like 
```
  selectOptions.contestStatus.filter((status) => statusOptions[contestType].includes(status))`
```